t/01-temp-files.t

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/perl
+# This program is open source, licensed under the PostgreSQL License.
+# For license terms, see the LICENSE file.
+#
+# Copyright (C) 2012-2022: Open PostgreSQL Monitoring Development Group
+
+use strict;
+use warnings;
+
+use lib 't/lib';
+use pgNode;
+use pgSession;
+use File::Path qw(rmtree);
+use Test::More tests => 39;
+
+my $node = pgNode->get_new_node('prod');
+my $proc;
+
+$node->init;
+$node->start;
+
+### Beginning of tests ###
+
+# This service can run without thresholds
+
+# Tests for PostreSQL 8.1 and before
+SKIP: {
+    skip "testing incompatibility with PostgreSQL 8.1 and before", 3
+        if $node->version >= 8.2;
+
+    $node->command_checks_all( [
+        './check_pgactivity', '--service'  => 'temp_files',
+                              '--username' => getlogin,
+                              '--format'   => 'human',
+        ],
+        1,
+        [ qr/^$/ ],
+        [ qr/^Service temp_files is not compatible with host/ ],
+        'non compatible PostgreSQL version'
+    );
+}
+
+SKIP: {
+    skip "incompatible tests with PostgreSQL < 8.2", 34 if $node->version < 8.2;
+
+    # basic check
+    $node->command_checks_all( [
+        './check_pgactivity', '--service'  => 'temp_files',
+                              '--username' => getlogin,
+                              '--format'   => 'human',
+                              '--dbname'   => 'template1',
+        ],
+        0,
+        [ qr/^Service  *: POSTGRES_TEMP_FILES$/m,
+          qr/^Returns  *: 0 \(OK\)$/m,
+          qr/^Message  *: 3 tablespace\(s\)\/database\(s\) checked$/m,
+        ],
+        [ qr/^$/ ],
+        'basic check'
+    );
+    sleep 2;
+
+    $proc = pgSession->new( $node, 'postgres' );
+
+    # unit test OK
+    $proc->query('SELECT random() * x FROM generate_series(1,1000000) AS F(x) ORDER BY 1;', 2);
```

Comment:
Maybe comment how much temp files are expected from this query?
How safe is it to build these tests without nailing down a `work_mem` and some `enable_*` setups to make sure the plan will always requires a controlled amount of temp file?

Reply:
```
[local]:5437 postgres@postgres=# EXPLAIN ANALYZE SELECT random() * x FROM generate_series(1,1000000) AS F(x) ORDER BY 1;,
                                                                QUERY PLAN                                                                
------------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=130832.35..133332.35 rows=1000000 width=8) (actual time=1542.076..1795.845 rows=1000000 loops=1)
   Sort Key: ((random() * (x)::double precision))
   Sort Method: external merge  Disk: 11768kB
   ->  Function Scan on generate_series f  (cost=0.00..17500.00 rows=1000000 width=8) (actual time=359.157..774.653 rows=1000000 loops=1)
 Planning Time: 4.357 ms
 JIT:
   Functions: 4
   Options: Inlining false, Optimization false, Expressions true, Deforming true
   Timing: Generation 0.674 ms, Inlining 0.000 ms, Optimization 3.605 ms, Emission 56.465 ms, Total 60.745 ms
 Execution Time: 1969.987 ms
(10 rows)
```

I don't think we need enable_*.
We could force work_mem in case the default changes (doesn't seem very likely ?).

With 11,7MB, I thought we were well within the boundaries I used. Should I make it bigger ?

| test | Warning  | Critical |
|-------|---------------|-----------|
| unit test OK | 3 | 4 |
| unit test WARN | 1 | 3 |
| unit test CRIT | 0 | 1 | 
| test with ts | NA | NA |
| basic check | NA | NA |

Reply:
On a 9.4 I get 17.5MB.

Reply:
I added a bunch more tests (3 for the file size and 3 for size and file count )

Reply:
excellent, thanks!